pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go

```diff
@@ -308,15 +307,46 @@ func getPinnedLibraries(registry string) []libInfo {
 	return res
 }
 
-// getLibrariesLanguageDetection runs process language auto-detection and returns languages to inject for APM Instrumentation.
-// The languages information is available in workloadmeta-store and attached on the pod's owner.
-func (w *Webhook) getLibrariesLanguageDetection(pod *corev1.Pod) []libInfo {
-	if config.Datadog().GetBool("admission_controller.auto_instrumentation.inject_auto_detected_libraries") {
-		// Use libraries returned by language detection for APM Instrumentation
-		return w.getAutoDetectedLibraries(pod)
-	}
+type libInfoLanguageDetection struct {
+	libs             []libInfo
+	injectionEnabled bool
 
-	return nil
+	injected bool
+}
+
+func (l *libInfoLanguageDetection) podMutator(v version) podMutator {
+	return podMutatorFunc(func(pod *corev1.Pod) error {
+		if l == nil || len(l.libs) == 0 || !v.usesInjector() || l.injected {
+			return nil
+		}
+
+		var libs []string
+		for _, lib := range l.libs {
+			libs = append(libs, string(lib.lang))
+		}
+
+		_ = mutatecommon.InjectEnv(pod, corev1.EnvVar{
+			Name:  "DD_INSTRUMENTATION_LANGUAGES_DETECTED",
+			Value: strings.Join(libs, ","),
+		})
+		_ = mutatecommon.InjectEnv(pod, corev1.EnvVar{
+			Name:  "DD_INSTRUMENTATION_LANGUAGE_DETECTION_INJECTION_ENABLED",
+			Value: strconv.FormatBool(l.injectionEnabled),
+		})
```

Comment:
Are you sure you don't need to propagate whether language detection was enabled or not?

If we don't propagate this information, then what would happen if the injector receives the following:
```
DD_INSTRUMENTATION_LANGUAGES_DETECTED=
DD_INSTRUMENTATION_LANGUAGE_DETECTION_INJECTION_ENABLED=false
```

This can imply one of the following:
- Language detection is not enabled in the first place
- Language detection is enabled, but for some reason, we detected nothing.

Without the knowledge of whether language detection is enabled or not, we will not be able to decide which of the two implications the correct one is. 

If I could convince you, I'd suggest two ways around this:
- Conditionally inject these 2 env vars only when `language_detection.enabled` and `language_detection.reporting.enabled` are `true`.
- Inject a third env var `DD_INSTRUMENTATION_LANGUAGE_DETECTION_ENABLED`